src/test/java/com/google/devtools/build/lib/remote/blobstore/http/HttpBlobStoreTest.java

```diff
@@ -54,73 +65,232 @@
 import io.netty.handler.timeout.ReadTimeoutException;
 import java.io.ByteArrayInputStream;
 import java.io.ByteArrayOutputStream;
+import java.io.File;
 import java.net.ConnectException;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
 import java.net.URI;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.function.IntFunction;
+import javax.annotation.Nullable;
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.junit.runners.JUnit4;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+import org.mockito.AdditionalAnswers;
 import org.mockito.Mockito;
 
-/** Tests for {@link HttpBlobStore}. */
-@RunWith(JUnit4.class)
+/**
+ * Tests for {@link HttpBlobStore}.
+ */
+@RunWith(Parameterized.class)
 public class HttpBlobStoreTest {
 
-  private ServerSocketChannel startServer(ChannelHandler handler) throws Exception {
-    EventLoopGroup eventLoop = new NioEventLoopGroup(1);
+  private static ServerChannel createServer(
+      Class<? extends ServerChannel> serverChannelClass,
+      IntFunction<EventLoopGroup> newEventLoopGroup,
+      SocketAddress socketAddress,
+      ChannelHandler handler) {
+    EventLoopGroup eventLoop = newEventLoopGroup.apply(1);
     ServerBootstrap sb =
         new ServerBootstrap()
             .group(eventLoop)
-            .channel(NioServerSocketChannel.class)
+            .channel(serverChannelClass)
             .childHandler(
-                new ChannelInitializer<NioSocketChannel>() {
+                new ChannelInitializer<Channel>() {
                   @Override
-                  protected void initChannel(NioSocketChannel ch) {
+                  protected void initChannel(Channel ch) {
                     ch.pipeline().addLast(new HttpServerCodec());
                     ch.pipeline().addLast(new HttpObjectAggregator(1000));
                     ch.pipeline().addLast(handler);
                   }
                 });
-    return ((ServerSocketChannel) sb.bind("localhost", 0).sync().channel());
+    try {
+      return ((ServerChannel) sb.bind(socketAddress).sync().channel());
+    } catch (Exception e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  private static DomainSocketAddress newDomainSocketAddress() {
+    try {
+      File file = File.createTempFile("bazel", ".sock", TestUtils.tmpDirFile());
```

Comment:
We found the problem to be the length of the filename. At least on Linux
there seems to be a 100 character limit for socket names. Can you please
replace the `TestUtils.tmpDirFile()` with `new File("/tmp/")` or so to ensure
shorter names.